From e4e078e2215e1a4c66e145fead0e25ed8d057905 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 9 Jan 2026 22:22:41 +0100
Subject: [PATCH] fix: Disable GCC colored diagnostics in test_CounterHistoFail

GCC with -fdiagnostics-color=always embeds ANSI escape codes in compiler
output. This breaks the regex match in the test_CounterHistoFail test
which checks for a specific error message in the compiler output.

Setting GCC_COLORS="" disables the colored output for this test.
---
 GaudiKernel/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/GaudiKernel/CMakeLists.txt b/GaudiKernel/CMakeLists.txt
index 5a591e1107..8ef3d4aa55 100644
--- a/GaudiKernel/CMakeLists.txt
+++ b/GaudiKernel/CMakeLists.txt
@@ -1,5 +1,5 @@
 #####################################################################################
-# (c) Copyright 1998-2025 CERN for the benefit of the LHCb and ATLAS collaborations #
+# (c) Copyright 1998-2026 CERN for the benefit of the LHCb and ATLAS collaborations #
 #                                                                                   #
 # This software is distributed under the terms of the Apache version 2 licence,     #
 # copied verbatim in the file "LICENSE".                                            #
@@ -383,6 +383,8 @@ if(BUILD_TESTING)
   add_test(NAME GaudiKernel.test_CounterHistoFail COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} --target test_CounterHistoFail)
   set_property(TEST GaudiKernel.test_CounterHistoFail PROPERTY PASS_REGULAR_EXPRESSION "\\+\\+h2d\\[1\\.2\\]\; // ERROR: cannot fill a 2D histogram with a single index")
   set_property(TEST GaudiKernel.test_CounterHistoFail PROPERTY RESOURCE_LOCK build-system)
+  # Disable GCC colored diagnostics to avoid ANSI escape codes in output breaking the regex match
+  set_property(TEST GaudiKernel.test_CounterHistoFail PROPERTY ENVIRONMENT "GCC_COLORS=")
 
   gaudi_add_executable(GaudiKernel_catch2_tests
     SOURCES
-- 
GitLab

